sims/verilator/Makefile

```diff
@@ -47,67 +47,133 @@ debug: $(sim_debug)
 include $(base_dir)/common.mk
 
 #########################################################################################
-# verilator binary and flags
+# verilator-specific user-interface variables and commands
 #########################################################################################
-VERILATOR := verilator --cc --exe
-
-CXXFLAGS := \
-	$(CXXFLAGS) -O1 -std=c++11 \
-	-I$(RISCV)/include \
-	-I$(dramsim_dir) \
-	-D__STDC_FORMAT_MACROS \
-	$(EXTRA_SIM_CC_FLAGS)
+HELP_COMPILATION_VARIABLES += \
+"   VERILATOR_PROFILE      = 'none' if no verilator profiling (default)" \
+"                            'all' if full verilator runtime profiling" \
+"                            'threads' if runtime thread profiling only" \
+"   VERILATOR_FST_MODE     = enable FST waveform instead of VCD. use with debug build"
 
-LDFLAGS := \
-	$(LDFLAGS) \
-	-L$(sim_dir) \
-	-lpthread
+#########################################################################################
+# verilator/cxx binary and flags
+#########################################################################################
+VERILATOR := verilator --cc --exe
 
-VERILATOR_CC_OPTS = \
+#----------------------------------------------------------------------------------------
+# user configs
+#----------------------------------------------------------------------------------------
+VERILATOR_PROFILE ?= none
+RUNTIME_PROFILING_CFLAGS := $(if $(filter $(VERILATOR_PROFILE),all),-g -pg,)
```

Comment:
Probably

Reply:
Actually, if you want this in one line then this is the good way to do it. Otherwise, you will have to do something like:

```
if (condition)
RUNTIME_PROFILING_CFLAGS := -g -pg
else
RUNTIME_PROFILING_CFLAGS :=
endif
```